src/fdb5/tools/fdb-hide.cc

```diff
@@ -83,10 +83,10 @@ void FdbHide::execute(const option::CmdArgs& args) {
 
     const Schema& schema = conf.schema();
 
-    Key dbkey;
+    TypedKey dbkey{schema.registry()};
     ASSERT(schema.expandFirstLevel(dbrequest.request(), dbkey));
 
-    std::unique_ptr<Catalogue> db = CatalogueFactory::instance().build(dbkey, conf, true);
+    std::unique_ptr<Catalogue> db = CatalogueFactory::instance().build(dbkey.canonical(), conf, true);
```

Comment:
db and dbWrited (below) could share a `const auto dbCanonicalKey = dbkey.canonical();`